Authentication/LogIn/View/LoginController.swift

```diff
@@ -0,0 +1,149 @@
+            if executing {
+                self._delegate.loginControllerWillExecuteLogIn(self)
+            } else {
+                self._delegate.loginControllerDidExecuteLogIn(self)
+            }
+        }
+        
+        _viewModel.logInErrors.observeNext { [unowned self] in self._delegate.loginController(self, didLogInWithError: $0) }
+        
+    }
+    
+    func bindEmailElements() {
+        _viewModel.email <~ loginView.emailTextField.rex_textSignal
+        loginView.emailLabel.text = _viewModel.emailText
+        loginView.emailTextField.placeholder = _viewModel.emailPlaceholderText
+        if let emailValidationMessageLabel = loginView.emailValidationMessageLabel {
+            emailValidationMessageLabel.rex_text <~ _viewModel.emailValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindPasswordElements() {
+        _viewModel.password <~ loginView.passwordTextField.rex_textSignal
+        loginView.passwordLabel.text = _viewModel.passwordText
+        loginView.passwordTextField.placeholder = _viewModel.passwordPlaceholderText
+        //loginView.passwordTextField.secureTextEntry = !_viewModel.showPassword.value    // initial value may be required, check with tests
+        if let passwordValidationMessageLabel = loginView.passwordValidationMessageLabel {
+            passwordValidationMessageLabel.rex_text <~ _viewModel.passwordValidationErrors.signal.map { $0.first ?? " " }
+        }
+        if let passwordVisibilityButton = loginView.passwordVisibilityButton {
+            //passwordVisibilityButton.setTitle(_viewModel.passwordVisibilityButtonTitle, forState: .Normal)  // initial value may be required, check with tests
+            passwordVisibilityButton.rex_title <~ _viewModel.showPassword.producer.map { [unowned self] _ in
+                self._viewModel.passwordVisibilityButtonTitle
+            }
+            passwordVisibilityButton.rex_pressed.value = _viewModel.togglePasswordVisibility.unsafeCocoaAction
+            _viewModel.showPassword.signal.observeNext { [unowned self] showPassword in
+                self.loginView.passwordTextField.secureTextEntry = !showPassword
+            }
+        }
+    }
+    
+    func bindButtons() {
+        loginView.logInButton.setTitle(_viewModel.loginButtonTitle, forState: .Normal)
+        loginView.logInButton.rex_pressed.value = _viewModel.logInCocoaAction
+        
+        loginView.registerButton.setTitle(_viewModel.registerButtonTitle, forState: .Normal)
+        loginView.registerButton.setAction { [unowned self] _ in self._onRegister(self) }
+        
+        // loginView.termsAndService -> Present modally web view controller that shows HTML file
+        loginView.termsAndService?.setTitle(_viewModel.termsAndServicesButtonTitle, forState: .Normal)
+    }
+    
+}
+
+public extension SessionServiceError {
+    var message: String {
+        switch self {
+        case .InexistentUser:
```

Comment:
We should merge `InexistentUser` and `WrongPassowd` into `InvalidCredentials(NSError?)`. For security reason server don't actually give the exact cause of error to avoid brute force attacks.